src/mjolnir/shortcutbuilder.cc

```diff
@@ -527,6 +531,20 @@ uint32_t AddShortcutEdges(GraphReader& reader,
       // Compute the weighted edge density
       newedge.set_density(average_density / (static_cast<float>(length)));
 
+      // Update speed to the one that takes turn durations into account
+      uint32_t new_speed = directededge->speed();
+      if (total_duration > 0) {
```

Comment:
maybe change these ifs to assertions since its insane that it would be 0

Reply:
Yeah if speed can't be 0 on the edges then this should be some meaningful positive value. Actually the only incorrect case I can imagine is when edge speed is zero and the should result in zero speed on the whole shortcut. But we'll have division by zero first...